app/addons/components/react-components.react.jsx

```diff
@@ -60,6 +62,94 @@ function (app, FauxtonAPI, React, Stores, FauxtonComponents, ace, beautifyHelper
     }
   });
 
+
+  var BulkActionComponent = React.createClass({
+
+    propTypes: {
+      hasSelectedItem: React.PropTypes.bool.isRequired,
+      removeItem: React.PropTypes.func.isRequired,
+      selectAll: React.PropTypes.func.isRequired,
+      toggleSelect: React.PropTypes.func.isRequired,
+      isChecked: React.PropTypes.bool.isRequired
+    },
+
+    getDefaultProps: function () {
+      return {
+        disabled: false,
+        title: 'Select rows that can be...',
+        bulkIcon: 'fonticon-trash',
+        buttonTitle: 'Delete all selected',
+        dropdownContentText: 'Deleted'
+      };
+    },
+
+    render: function () {
+      return (
+        <div className="bulk-action-component">
+          <div className="bulk-action-component-selector-group">
+            {this.getMasterSelector()}
+            {this.getMultiSelectOptions()}
+          </div>
+        </div>
+      );
+    },
+
+    getMultiSelectOptions: function () {
+      if (!this.props.hasSelectedItem) {
+        return null;
+      }
+
+      return (
+        <div className="group-clean bulk-actions panel">
+          <button
+            onClick={this.props.removeItem}
+            className={'fonticon ' + this.props.bulkIcon}
+            title={this.props.buttonTitle} />
+        </div>
+      );
+    },
+
+    getPopupContent: function () {
+      return (
+        <ul className="bulk-action-component-popover-actions">
+          <li onClick={this.selectAll} >
+            <i className="icon fonticon-cancel"></i> {this.props.dropdownContentText}
+          </li>
+        </ul>
+      );
+    },
+
+    selectAll: function () {
+      this.refs.popover.hide();
+      this.props.selectAll();
+    },
+
+    getMasterSelector: function () {
+      return (
+        <div className="group-clean bulk-actions-panel panel">
+          <input type="checkbox"
+            checked={this.props.isChecked}
+            onChange={this.props.toggleSelect}
+            disabled={this.props.disabled} />
+          <div className="separator"></div>
+          <ReactBootstrap.OverlayTrigger
+            ref='popover'
```

Comment:
i don't quite understand, in case you meant that the name is too generic, i changed it too `bulkActionPopover` now